File: packages/shared/src/enums.ts
export const workModes = ["remote", "hybrid", "onsite"] as const;
export const employmentTypes = ["full_time", "contract", "internship"] as const;
export const jobSources = ["greenhouse", "lever", "ashby", "manual"] as const;
export const matchStrengths = ["high", "medium", "low"] as const;
export const sourceHealthStatuses = ["healthy", "warning", "inactive"] as const;
export const reviewActions = ["approve", "reject"] as const;
export const applicationStatuses = [
  "discovered",
  "shortlisted",
  "draft_ready",
  "needs_user_input",
  "waiting_review",
  "submitting",
  "submitted",
  "failed",
  "blocked",
  "duplicate"
] as const;
export const packetStatuses = [
  "draft_ready",
  "needs_user_input",
  "waiting_review",
  "finalized"
] as const;

export type WorkMode = (typeof workModes)[number];
export type EmploymentType = (typeof employmentTypes)[number];
export type JobSource = (typeof jobSources)[number];
export type MatchStrength = (typeof matchStrengths)[number];
export type SourceHealthStatus = (typeof sourceHealthStatuses)[number];
export type ReviewAction = (typeof reviewActions)[number];
export type ApplicationStatus = (typeof applicationStatuses)[number];
export type PacketStatus = (typeof packetStatuses)[number];

File: packages/shared/python/job_focus_shared/enums.py
from enum import StrEnum


class WorkMode(StrEnum):
    REMOTE = "remote"
    HYBRID = "hybrid"
    ONSITE = "onsite"


class EmploymentType(StrEnum):
    FULL_TIME = "full_time"
    CONTRACT = "contract"
    INTERNSHIP = "internship"


class JobSource(StrEnum):
    GREENHOUSE = "greenhouse"
    LEVER = "lever"
    ASHBY = "ashby"
    MANUAL = "manual"


class MatchStrength(StrEnum):
    HIGH = "high"
    MEDIUM = "medium"
    LOW = "low"


class SourceHealthStatus(StrEnum):
    HEALTHY = "healthy"
    WARNING = "warning"
    INACTIVE = "inactive"


class ReviewAction(StrEnum):
    APPROVE = "approve"
    REJECT = "reject"


class ApplicationStatus(StrEnum):
    DISCOVERED = "discovered"
    SHORTLISTED = "shortlisted"
    DRAFT_READY = "draft_ready"
    NEEDS_USER_INPUT = "needs_user_input"
    WAITING_REVIEW = "waiting_review"
    SUBMITTING = "submitting"
    SUBMITTED = "submitted"
    FAILED = "failed"
    BLOCKED = "blocked"
    DUPLICATE = "duplicate"


class PacketStatus(StrEnum):
    DRAFT_READY = "draft_ready"
    NEEDS_USER_INPUT = "needs_user_input"
    WAITING_REVIEW = "waiting_review"
    FINALIZED = "finalized"
